### PR TITLE
Add size, created_at, and download to the attachments API

### PR DESCRIPTION
**Pull request by aapomm**
### Summary

Currently, when users query the API for attachments, our API returns the file name and location rather than the file itself.

*Proposed Solution*
- Update the #show Attachments endpoint to return the size, created_at, and download link
- Implement a #download Attachments endpoint for serving the actual attachment files.

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [ ] Added a CHANGELOG entry

### CODE REVIEW

File: engines/dradis-api/spec/requests/dradis/ce/api/v3/attachments_spec.rb [hunk 1]
```diff
@@ -289,5 +300,17 @@
           'Resource deleted successfully'
       end
     end
+
+    describe 'GET /api/nodes/:node_id/attachments/:filename/download' do
+      it 'responds with the file' do
```
Comment (caitmich) [inline engines/dradis-api/spec/requests/dradis/ce/api/v3/attachments_spec.rb:305]: returns/downloads the file?

File: engines/dradis-api/app/controllers/dradis/ce/api/v3/attachments_controller.rb [hunk 2]
```diff
@@ -71,6 +73,15 @@ def destroy
         render_successful_destroy_message
       end
 
+      def download
+        begin
+          @attachment = Attachment.find(params[:filename], conditions: { node_id: @node.id })
+          send_file(@attachment.fullpath)
```
Comment (caitmich) [inline engines/dradis-api/app/controllers/dradis/ce/api/v3/attachments_controller.rb:79]:
```suggestion
          send_file(@attachment.fullpath, disposition: 'attachment')
```

File: engines/dradis-api/app/controllers/dradis/ce/api/v3/attachments_controller.rb [hunk 3]
```diff
@@ -15,6 +15,8 @@ def index
       def show
         begin
           @attachment = Attachment.find(params[:filename], conditions: { node_id: @node.id })
+          @file_size = File.size(@attachment.fullpath)
+          @created_at = File.ctime(@attachment.fullpath)
```
Comment (etdsoft) [inline engines/dradis-api/app/controllers/dradis/ce/api/v3/attachments_controller.rb:19]: Why are these needed?

```ruby
class Attachment < File
```
